### ISSUE TITLE
Why there is no corresponding .cpp file for dcgm_agent.h?

### ISSUE THREAD

**Issue by irvingans** [NVIDIA/DCGM#99]
https://github.com/NVIDIA/DCGM/blob/7e1012302679e4bb7496483b32dcffb56e528c92/dcgmlib/dcgm_agent.h#L1228

Hi,
I am looking for the specific implemetation method for dcgmGetPidInfo(),   but I can not find the corresponding .cpp file for dcgm_agent.h. Why is this?

**Comment by nikkon-dev** [comment]
The dcgm_agent.h is the public API header and is not directly related to the implementation part. 
Take a look [here](https://github.com/NVIDIA/DCGM/blob/7e1012302679e4bb7496483b32dcffb56e528c92/dcgmlib/entry_point.h#L679) and look for the tsapiEngineGetPidInfo in the [DcgmApi.cpp](https://github.com/NVIDIA/DCGM/blob/7e1012302679e4bb7496483b32dcffb56e528c92/dcgmlib/src/DcgmApi.cpp#L3977) file.

**Comment by irvingans** [comment]
Hi @nikkon-dev , in the [DcgmApi.cpp](https://github.com/NVIDIA/DCGM/blob/7e1012302679e4bb7496483b32dcffb56e528c92/dcgmlib/src/DcgmApi.cpp#L3977) file, tsapiEngineGetPidInfo(dcgmHandle_t pDcgmHandle, dcgmGpuGrp_t groupId, dcgmPidInfo_t *pidInfo) ,  pidInfo is what users pass in instead of being obtained from some other sources, right?

**Comment by nikkon-dev** [comment]
Yes, that's what the user passes to the function setting two fields: version (dcgmPidInfo_version2 for now) and pid, which is the PID of a desired process created ***after*** the stats gathering was enabled.

**Comment by optyang** [comment]
> Yes, that's what the user passes to the function setting two fields: version (dcgmPidInfo_version2 for now) and pid, which is the PID of a desired process created _**after**_ the stats gathering was enabled.

Hi @nikkon-dev , how to enable *stats gathering*? Thanks!
